### PR TITLE
Incorporating WG feedback

### LINKED ISSUE #1187: Creation Time and Expiration Time do not allow for clock skew
The processing instructions for Creation Time and Expiration Time imply that verifiers are not permitted to account for clock skew during signature verification.

### LINKED ISSUE #1188: Require lowercased header field names as identifiers
The current text allows mixed-case header field names when they are being used as content identifiers. This is unnecessary – header field names are case-insensitive – and creates opportunity for incompatibility.  Instead, content identifiers should always be lowercase.

### PR DESCRIPTION
**Pull request by richanna**
- Strengthened requirement for content identifiers for header fields to be lower-case (changed from SHOULD to MUST).
- Relaxed guidance on processing Creation Time and Expiration Time to only require verifiers to examine them and test against the verifier’s requirements.
- Minor editorial corrections and readability improvements.

Resolves #1187, resolves #1188

### CODE REVIEW

File: draft-ietf-httpbis-message-signatures.md [hunk 1]
```diff
@@ -401,7 +406,7 @@ In order to verify a signature, a verifier MUST:
 3. Use the signature's Algorithm and Verification Key Material with the recreated Signing Input to verify the signature value.
 
 
-A signature with a Creation Time that is in the future or an Expiration Time that is in the past MUST NOT be processed.  
+The verifier MUST examine the signature's Creation Time and Expiration Time and evaluate whether they meet the verifier's requirements.  These requirements may vary depending on the use case.  When comparing Creation Time and/or Expiration Time to a timestamp obtained from another source (e.g., testing that the Expiration Time is later than the local time at the verifier), the verifier MAY allow for some variability due to clock skew between the two timestamp sources (e.g., the signer and the verifier).  The verifier MAY impose its own maximum age limits, and reject signatures that are too old (i.e., have a Creation Time too far in the past) regardless of whether the signer gave the signatures an Expiration Time.
```
Comment (ioggstream) [inline draft-ietf-httpbis-message-signatures.md:409]: I would split this passage in a separate PR and provide considerations. The rest LGTM.

Comment (jricher) [inline draft-ietf-httpbis-message-signatures.md:409]:
```suggestion
A signature with a Creation Time that is in the future or an Expiration Time that is in the past MUST NOT be processed.  
```

Comment (jricher) [reply draft-ietf-httpbis-message-signatures.md]: Done, now available in #1280 as a starting point
